src/software/util/typename/typename.h

```diff
@@ -0,0 +1,11 @@
+#include <cxxabi.h>
+
+/**
+ * MACRO to get typename of the object, i.e.
+ * ClassA object_a;
+ * TYPENAME(object_a); // returns ClassA
+ *
+ * @param object The object to get typename for
```

Comment:
Just one quick question, is this compiler dependent? Is this specific to GCC / Clang, or to newer versions of one or the other? (not that I expect to downgrade the compiler).

Reply:
cxxabi should be available in clang and gcc, but for clang there might be a bit more work https://stackoverflow.com/questions/19329525/how-to-demangle-a-c-name-in-clang-or-gcc. There could be some issues downgrading the compiler so `-D_GLIBCXX_USE_CXX11_ABI` might be needed

Reply:
As long as it's doable in both compilers, and not something super specific (like that clang-specific magic enum stuff we tried), I'm not too concerned. I also don't see us ever downgrading, so that shouldn't be an issue either.

Reply:
I'm pretty sure this leaks memory. From [libstdc++](https://gcc.gnu.org/onlinedocs/libstdc++/libstdc++-html-USERS-4.3/a01696.html):
>  A pointer to the start of the NUL-terminated demangled name, or NULL if the demangling fails. The caller is responsible for deallocating this memory using free.

Reply:
oh wow, great catch, it did leak memory. I've fixed it now